### PR TITLE
Add S07 and S10 send API tests with event invariants helper 

### PR DESCRIPTION
**Pull request by at0m1x19**
- Add S07 test: core sender with relay + store peer, reliability enabled. Verifies `Propagated` → `Sent` event sequence.                                                                                                                                                                                         
- Add S10 test: edge sender with lightpush-only path (3-node topology: edge sender → service node → relay peer). Verifies `Propagated` only.
- Add `assert_event_invariants` helper to enforce per-request event constraints from issue #163 spec.                                                                                                                                                                                                            
- Integrate the invariants helper into all existing tests (S06, S17, S19, S21) and both new tests.                                                                                                                                                                                                               
                     

Test plan                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                   
  - [x] S07 passes locally and in CI                                                                                                                                                                                                                                                                                    
  - [x] S10 passes locally and in CI         
  - [x] All pre-existing tests (S06, S17×2, S19, S21) pass locally and in CI         

## Issue
- https://github.com/logos-messaging/logos-delivery-interop-tests/issues/163

### CODE REVIEW

File: src/node/wrapper_helpers.py [hunk 1]
```diff
@@ -88,6 +88,41 @@ def wait_for_error(collector: EventCollector, request_id: str, timeout_s: float)
...
+        assert event.get("requestId") == request_id, (
+            f"Event at index {i} has wrong requestId: " f"expected {request_id!r}, got {event.get('requestId')!r}"
+        )
+        event_type = event.get("eventType", "")
+        if event_type in TERMINAL_EVENT_TYPES:
+            counts[event_type] = counts.get(event_type, 0) + 1
+            if event_type not in first_index:
+                first_index[event_type] = i
+
+    for event_type, count in counts.items():
```
Comment (fbarbu15) [inline src/node/wrapper_helpers.py:115]: don't we need an `assert counts` for the case where counts is {} ?

**Review by fbarbu15 — APPROVED** [review]

File: src/node/wrapper_helpers.py (hunk 1 above)
Comment (at0m1x19) [reply src/node/wrapper_helpers.py:115]: counts can't be empty here, assert_event_invariants is always called after a wait_for_propagated/wait_for_sent/wait_for_error that already found a terminal event in the collector. 

That event is in events, so it will be counted.

**Review by AYAHASSAN287 — APPROVED** [review]
